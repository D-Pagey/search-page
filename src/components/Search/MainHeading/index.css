.main-heading-container {
  padding: 15px 0;
}

.main-heading {
  line-height: 1.1;
  margin: 0;
}