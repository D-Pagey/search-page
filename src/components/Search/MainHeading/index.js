import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const MainHeadingWrapper = styled.div`
  padding: 15px 0;
`;

const MainHeadingTitle = styled.h2`
  line-height: 1.1;
  margin: 0;
`;

export default function MainHeading({ total }) {
  return (
    <MainHeadingWrapper>
      <MainHeadingTitle>
        {total} VEHICLES FOUND NEAR LONDON, UNITED KINGDOM
      </MainHeadingTitle>
    </MainHeadingWrapper>
  );
}

MainHeading.propTypes = {
  total: PropTypes.number,
};

MainHeading.defaultProps = {
  total: '',
};
