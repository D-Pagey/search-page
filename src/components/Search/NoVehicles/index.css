.no-vehicles-container {
    background-color: white;
    border: 1px solid #e4e3e3;
    border-radius: 6px;
    padding: 16px;
    width: 290px;
}

.no-vehicles-title {
    font-size: 28px;
    letter-spacing: -1.2px;
    line-height: 1.1;
    margin: 0;
    text-align: center;
    text-transform: uppercase;
}

.no-vehicles-form {
    display: flex;
    flex-direction: column;
}

.form-group {
    display: flex;
    flex-direction: column;
    margin-bottom: 10px;
}

.no-vehicles-label {
    color: black;
    font-size: 16px;
    font-weight: bold;
    margin-bottom: 0.2rem;
}

.form-input {
    height: 40px;
    padding: 0 10px;
    border: 1px solid #e4e3e3;
    border-radius: 6px;
    font-size: 1rem;
}

.row {
    flex-direction: row;
}

.checkbox { 
    height: 50px;
    width: 50px;
}

.updates-copy {
    padding-left: 20px;
}