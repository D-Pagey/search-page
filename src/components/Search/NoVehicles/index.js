import React from 'react';

import './index.css';
import Button from '../../Button';

export default function NoVehicles() {
  return (
    <div className="no-vehicles-container">
      <div>
        <h3 className="no-vehicles-title">looking for a specific car?</h3>
        <p>Let us know what car you are looking for, and we&apos;ll be in touch!</p>
      </div>
      <form className="no-vehicles-form">
        <div className="form-group">
          <label htmlFor="fullname" className="no-vehicles-label">Full Name</label>
          <input
            type="text"
            required
            id="fullname"
            name="fullname"
            className="form-input"
            placeholder="John Doe"
          />
        </div>
        <div className="form-group">
          <label htmlFor="email" className="no-vehicles-label">Email address</label>
          <input
            type="email"
            required
            id="email"
            name="email"
            className="form-input"
            placeholder="example@gmail.com"
          />
        </div>
        <div className="form-group">
          <label htmlFor="time-frame" className="no-vehicles-label">
          When do you want a car?
          </label>
          <select
            type="text"
            required
            id="time-frame"
            name="time-frame"
            className="form-input"
            placeholder="example@gmail.com"
          >
            <option value="this-month">This Month</option>
            <option value="3-months">Within 3 Months</option>
            <option value="6-months">Within 6 Months</option>
            <option value="more-6-months">6 Months +</option>
          </select>
        </div>
        <div className="form-group">
          <label htmlFor="notes" className="no-vehicles-label">
          What are you looking for?
          </label>
          <input
            type="text"
            id="notes"
            name="notes"
            className="form-input"
            placeholder="A White BMW M3"
          />
        </div>
        <div className="form-group row">
          <input
            type="checkbox"
            id="checkbox"
            name="checkbox"
            className="checkbox"
          />
          <label htmlFor="checkbox" className="updates-copy">I would like to
          receive updates about discounts, special offers, news and events from Drover.
          </label>
        </div>
        <Button text="Find my car" />
      </form>
    </div>
  );
}
