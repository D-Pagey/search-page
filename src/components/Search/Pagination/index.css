.pagination-container {
  align-items: center;
  flex-direction: column;
  display: flex;
  margin-bottom: 1rem;
  width: 290px;
}

.pagination-results {
  margin: 0;
}

.pagination-btn-container {
  height: 40px;
  margin-top: 10px;
  width: 200px;
}

.page-list {
  display: flex;
  height: inherit;
  justify-content: space-evenly;
  list-style: none;
  margin: 0;
  padding: 0;
}

.page-item {
  display: flex;
}

.page-btn {
  background-color: white;
  border: 0;
  border-radius: 6px;
  padding: 10px;
}

.active {
  background-color: #50ff7d;
}