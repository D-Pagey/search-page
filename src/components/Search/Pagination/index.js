import React from 'react';
import { func, objectOf, any } from 'prop-types';

import './index.css';

export default function Pagination({ handleChange, userParams, totalCount }) {
  const { page } = userParams;

  return (
    <div className="pagination-container">
      <p className="pagination-results">
        {`Showing ${page}-${page}0 of
        ${totalCount ? totalCount.total_count : 0} results`}
      </p>
      <div className="pagination-btn-container">
        <ul className="page-list">
          <li className="page-item">
            <button
              type="button"
              className={`page-btn ${userParams.page === 1 ? 'active' : null}`}
              name="page"
              value="1"
              onClick={handleChange}
            >
            1
            </button>
          </li>
          <li className="page-item">
            <button
              type="button"
              className={`page-btn ${userParams.page === 2 ? 'active' : null}`}
              name="page"
              value="2"
              onClick={handleChange}
            >
            2
            </button>
          </li>
          <li className="page-item">
            <button
              type="button"
              className={`page-btn ${userParams.page === 3 ? 'active' : null}`}
              name="page"
              value="3"
              onClick={handleChange}
            >
            3
            </button>
          </li>
          <li className="page-item">
            <button
              type="button"
              className={`page-btn ${userParams.page === 4 ? 'active' : null}`}
              name="page"
              value="4"
              onClick={handleChange}
            >
            4
            </button>
          </li>
          <li className="page-item">
            <button
              type="button"
              className={`page-btn ${userParams.page === 5 ? 'active' : null}`}
              name="page"
              value="5"
              onClick={handleChange}
            >
            5
            </button>
          </li>
        </ul>
      </div>
    </div>
  );
}

Pagination.propTypes = {
  handleChange: func.isRequired,
  userParams: objectOf(any).isRequired,
  totalCount: objectOf(any),
};

Pagination.defaultProps = {
  totalCount: {},
};

/** To Do
 * Create page buttons based off of total results
 * Trigger rerender the page based off page number
 */
