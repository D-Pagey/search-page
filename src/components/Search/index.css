.search-container {
  align-items: center;
  display: flex;
  flex-direction: column;
  padding: 0 15px;
}