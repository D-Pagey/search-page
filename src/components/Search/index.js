import React, { Component } from 'react';

import './index.css';
import Banner from './Banner';
import MainHeading from './MainHeading';
import SliderBlock from './SliderBlock';
import VehicleList from './VehicleList';
import Pagination from './Pagination';
import NoVehicles from './NoVehicles';

const URL = 'https://app.joindrover.com/api/web/vehicles?vehicle_type=Consumer';

export default class Search extends Component {
  state = {
    results: [],
  }

  componentDidMount() {
    this.fetchData();
  }

  fetchData = () => {
    fetch(URL, {
      method: 'POST',
      headers: {
        'content-type': 'application/json',
      },
    })
      .then(response => response.json())
      .then(data => this.setState({
        results: data,
      }));
  }

  render() {
    const { results } = this.state;

    return (
      <main className="search-container">
        <Banner />
        <MainHeading count={results.metadata} />
        <SliderBlock submit={this.fetchData} />
        <VehicleList results={results} />
        <Pagination />
        <NoVehicles />
      </main>
    );
  }
}
