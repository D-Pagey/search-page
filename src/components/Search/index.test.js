import React from 'react';
import { shallow } from 'enzyme';
import Search from '.';

it('should render', () => {
  const wrapper = shallow(<Search />);
  expect(wrapper).toMatchSnapshot();
});
