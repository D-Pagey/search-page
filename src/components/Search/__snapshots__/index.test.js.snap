// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<main>
  <MainHeading
    count={Object {}}
  />
  <SliderBlock />
  <VehicleList />
  <Pagination
    metadata={Object {}}
  />
  <NoVehicles />
</main>
`;
